### DISCUSSION TITLE
template that automates up:: [[parent note]]

### DISCUSSION THREAD

**Discussion by 081isaac** [SilentVoid13/Templater#1173] [Ideas & Feature Requests]
I follow a note organization system from Linking Your Thinking, and at the top of his template he has a place to link the parent note to better visualize in graph view and to connect the notes.

Example, a parent note could be technology. Then you could create a link to a new note inside of the technology note called phone. Inside of the phone note he applies a template that has "up:: " written at the top. But, every time he has to manually type in the parent note, in this case it would like "up:: [[technology]]". Is there a way to automate the linking of the parent note for the template?

**Comment by Zachatoo** [comment]
What's the user flow? Are you clicking on a link to a note that doesn't exist, which is triggering a template that sets the `up:: [[parent]]` for you? If you aren't, then how do you know what the parent of that note is?

**Reply by 081isaac** [reply]
Yes, you click on a link to a note that does not exist. Since I have the Hotkeys for templates plugin I then normally just press ctrl+t (the hotkey I designated for this template). This creates the "up:: " text for me, but then I have to manually go up and type the link to the parent note. And, idk the coding part for this, but I feel like there should be a way in obsidian to find what note is linked to the newly created note.

**Reply by Zachatoo** [reply]
So really, what you want is a link to the note that was previously open before you opened this new note and applied a template to it. Does that sound right?

**Reply by danielo515** [reply]
Why not doing it in reverse order? To trigger the child note creation from the parent note, so you just need to insert a link to the previous file.

**Comment by 081isaac** [comment]
Yes.

**Reply by Zachatoo** [reply]
~~I don't know if this will work consistently, but this worked in my limited testing.~~

```js
up:: [[<% app.workspace.lastActiveFile.basename %>]]
```

**Comment by Zachatoo** [comment]
I think I've found a better answer than my previous answer:

```js
<%*
const parentFile = tp.file.find_tfile(app.workspace.getLastOpenFiles()[0]);
const parentLink = app.fileManager.generateMarkdownLink(parentFile, tp.file.folder(true));
-%>
up:: <% parentLink %>
```

Will keep it updated [here](https://zachyoung.dev/posts/templater-snippets#generating-links) if I find anything better.

**Comment by 081isaac** [comment]
haha awesome man, this works. do you have like a crypto address for me to
compensate you
